tiatoolbox/tools/registration/wsi_registration.py

```diff
@@ -1059,3 +1064,146 @@ def register(
         image_transform = inverse_translation @ tissue_transform @ forward_translation
 
         return image_transform @ transform_initializer
+
+
+def estimate_bspline_transform(
+    fixed_image: np.ndarray,
+    moving_image: np.ndarray,
+    fixed_mask: np.ndarray,
+    moving_mask: np.ndarray,
+    grid_space: float = 50.0,
+    scale_factors: list = (1, 2, 5),
+    shrink_factor: list = (4, 2, 1),
+    smooth_sigmas: list = (4, 2, 1),
+    num_iterations: int = 100,
+    sampling_percent: float = 0.2,
+):
+    """Estimate BSpline transformation.
+
+    This function performs registration using multi-resolution Bspline approach.
+
+    Args:
+        fixed_image (:class:`numpy.ndarray`):
```

Comment:
If we can replace some of these by kwargs that would be great.

Reply:
What do you mean here? Everything from grid_space down is an optional arg or kwarg.

Reply:
https://github.com/TissueImageAnalytics/tiatoolbox/blob/1760d595678752b932649eaebbf677d9814bdd5a/tiatoolbox/tools/registration/wsi_registration.py#L1066-L1116